contracts/QuestFactory.sol

```diff
@@ -111,13 +123,25 @@ contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgrade
             rewardAmount_,
             questId_,
             address(rabbitHoleReceiptContract),
-            questFee,
+            protocolFee,
             protocolFeeRecipient
         );
 
         return newQuest;
     }
 
+    // do we require or return questfee when maxUses are up? -> currently return qusetfee
+    // will another person be creating quests on behalf of the owner of the discount token? if so the require below will need to change -> currently not possible
+    function doDiscountedFee(uint tokenId_) internal returns (uint16) {
+        require(questTerminalKeyContract.ownerOf(tokenId_) == msg.sender, "QuestFactory: caller is not owner of discount token");
+
+        (uint16 discountPercentage, uint16 maxDiscountUses, uint16 usedCount) = questTerminalKeyContract.discounts(tokenId_);
+        if(usedCount > maxDiscountUses) return questFee;
```

Comment:
I think this might need to be `>=` (please verify!)

Lets say we set `maxUses` to 1 when we mint the QuestTerminalKey NFT
- We deploy a quest -- now `usedCount` is set to 1
- We can still deploy another discounted quest because `usedCount` is 1 and `maxDiscountUses` is also 1